source/main/gameplay/RoRFrameListener.cpp

```diff
@@ -1621,6 +1615,8 @@ void SimController::UpdateSimulation(float dt)
 
             m_gfx_scene.BufferSimulationData();
 
+            m_actor_manager.UpdateAirbrakeInput(dt);
+            m_actor_manager.UpdateActorVisuals(dt, m_player_actor); // Needs to happen after BufferSimulationData()

```

Comment:
> // Needs to happen after BufferSimulationData()

May be temporarily true but only as a workaround. The function should really update simulation state of visual things (flares for example) and as such it should happen before `BufferSimData()`.

Reply:
> as such it should happen before `BufferSimData()`.

But why would we want to run `UpdateActorVisuals()` on outdated sim buffer data?
At this point of the code the physics thread does not run anyway.
I must be missing something.

Reply:
I mean the whole point of the SimBuffer is that we can run the visuals update while physics thread is running. Or not?

Reply:
I just actualy looked into the function -> 😲 😵 😫  it turned out I had no idea what's in there. I must have lost a commit somewhere. I apologize for confusing you.

What happened is that I wrote the 'Rods' + 'debugviews' remakes before I added SimBuffer, so it originally read data directly. I clearly remember putting the UpdateRods() etc... to GfxActor/GfxScene, but I guess I was rebasing too wildly and dropped that commit by accident 😊

Anyway, that function needs to be renamed - it's not supposed to update any Gfx, only related things in simulation like flare states (TODO: is it necessary? I left it as such to avoid touching multiplayer)

Reply:
I have a pending cleanup branch that builds on top of this pull request: https://github.com/ulteq/rigs-of-rods/commit/134381d95e29d9fc3469752bac55c3b7633dcdb9

My suggestion would be to merge both, then do the `UpdateActorVisuals()` cleanup.

Reply:
I'm after the 'ideal solution' as you illustrated above, but there's still legacy code which mixes it up, and I think we can postpone it after we merge https://github.com/RigsOfRods/rigs-of-rods/pull/1538